### PR TITLE
[Enhancement] Adjust calculation of OverallThroughput

### PR DESCRIPTION
**Pull request by liuyehcf**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

Change the way of calculating `OverallThroughput`, the difference is that the way time is calculated has changed

* Before: Using `NetworkTime`, which only contains the real network time
* This pr: The time interval between sending the first packet and receiving the last packet

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by liuyehcf** [comment]
run starrocks_admit_test

File: be/src/exec/pipeline/exchange/sink_buffer.cpp [hunk 1]
```diff
@@ -174,6 +182,7 @@ void SinkBuffer::cancel_one_sinker() {
 
 void SinkBuffer::_update_network_time(const TUniqueId& instance_id, const int64_t send_timestamp,
                                       const int64_t receive_timestamp) {
+    _last_receive_time = MonotonicNanos();
```
Comment (silverbullet233) [inline be/src/exec/pipeline/exchange/sink_buffer.cpp:185]: the last request must contain eos, maybe we needn't update _last_receive_time every time.

Comment (liuyehcf) [reply be/src/exec/pipeline/exchange/sink_buffer.cpp:185]: Emm, you cannot determine whether a result package is a eos or not. And seems unnecessary to add a struct to save the excess info

**Comment by liuyehcf** [comment]
run starrocks_clang-tidy

**Review by silverbullet233 — DISMISSED** [review]

File: be/src/exec/pipeline/exchange/sink_buffer.h [hunk 2]
```diff
@@ -165,6 +165,9 @@ class SinkBuffer {
     int64_t _pending_timestamp = -1;
     mutable std::atomic<int64_t> _last_full_timestamp = -1;
     mutable std::atomic<int64_t> _full_time = 0;
+
+    std::atomic<int64_t> _first_send_time = -1;
```
Comment (kangkaisen) [inline be/src/exec/pipeline/exchange/sink_buffer.h:169]: Do we really need to use atomic variable for profile counters? I think for profile counters, using simple variable is OK.

Comment (liuyehcf) [reply be/src/exec/pipeline/exchange/sink_buffer.h:169]: Yes, it is, inconsistency is acceptable

**Comment by liuyehcf** [comment]
run starrocks_admit_test

**Review by silverbullet233 — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

File: be/src/exec/pipeline/exchange/sink_buffer.cpp [hunk 3]
```diff
@@ -124,7 +124,9 @@ void SinkBuffer::update_profile(RuntimeProfile* profile) {
 
     auto* network_timer = ADD_TIMER(profile, "NetworkTime");
     auto* wait_timer = ADD_TIMER(profile, "WaitTime");
+    auto* overall_timer = ADD_TIMER(profile, "OverallTime");
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply be/src/exec/pipeline/exchange/sink_buffer.cpp:127]: Same as `OverallThroughput`, the time interval between the sending timing of the first packet and the receiving timing fo the last packet

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.2

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.4

**Pull request merged**